src/modules/data_module.py

```diff
@@ -92,7 +92,39 @@ def __init__(self, source, train_subset_size, test_subset_size,
         # an index 'i' will always refer to the exact same sample
         # this is necessary for easier evaluation later on
         train_all_idx = np.arange(train_size)
-        train_subset_indices = np.random.choice(train_all_idx, size=train_subset_size, replace=False)
+        if balanced_split:
+            classes_idx = {}
+            for target in range(len(self._full_train_set.classes)):
+                idx = np.where(self._full_train_set.targets == target)[0]
+                classes_idx[target] = idx
+
+            sizes = {}
+            if sum(ratio_classes) != 1:
+                raise ValueError("Ratios of the classes should sum to 1")
+            
+            for i, ratio in enumerate(ratio_classes):
+                class_size = int(ratio*train_subset_size)
+                sizes[i] = class_size
+
+            classes_idx_subsets = {}
+            for key in classes_idx:
+                class_indexes = classes_idx[key]
```

Comment:
I think you can use
`for key, value in classes_idx.items()`
to avoid using `class_indexes` variable (`value` will be the same thing)

You can rename `value` to `class_idx` as it's a better name